contrib/babelfishpg_tsql/src/procedures.c

```diff
@@ -2062,3 +2066,187 @@ gen_sp_droprolemember_subcmds(const char *user, const char *member)
+
+		/* if no function is found */
+		if(candidates == NULL)
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_OBJECT),
+					errmsg("function \"%s\" does not exist", psprintf("%s.%s", logical_schema_name, function_name))));
+	}
+	if(volatility == NULL)
+	{	
+		if(function_name == NULL)
+		{	
+			query = psprintf(
+					"SELECT t3.orig_name as SchemaName, t1.proname as FunctionName, "
+					"CASE "
+						"WHEN t1.provolatile = 'v' THEN 'volatile' "
+						"WHEN t1.provolatile = 's' THEN 'stable' "
+						"ELSE 'immutable' "
+					"END AS Volatility "
+					"from pg_proc t1 "
+					"JOIN pg_namespace t2 ON t1.pronamespace = t2.oid "
+					"JOIN sys.babelfish_namespace_ext t3 ON t3.nspname = t2.nspname "
+					"where t1.prokind = 'f' AND t3.dbid = sys.db_id('%s') ORDER BY t2.nspname, t1.proname", db_name
+				);
+		}
+		else
+		{
+			query = psprintf(
+					"SELECT '%s' as SchemaName, proname as FunctionName, "
+					"CASE "
+						"WHEN provolatile = 'v' THEN 'volatile' "
+						"WHEN provolatile = 's' THEN 'stable' "
+						"ELSE 'immutable' "
+					"END AS Volatility "
+					"from pg_proc "
+					"where oid = %u", logical_schema_name, candidates->oid
+				);
+		}
+
+		PG_TRY();
+		{	
+			char nulls = 0;
+			MemoryContext savedPortalCxt;
+			SPIPlanPtr plan;
+			Portal portal;
+			DestReceiver *receiver;
+
+			savedPortalCxt = PortalContext;
+			if (PortalContext == NULL)
+				PortalContext = MessageContext;
+			if ((rc = SPI_connect()) != SPI_OK_CONNECT)
+				elog(ERROR, "SPI_connect failed: %s", SPI_result_code_string(rc));
+			PortalContext = savedPortalCxt;
+
+			if ((plan = SPI_prepare(query, 0, NULL)) == NULL)
+				elog(ERROR, "SPI_prepare(\"%s\") failed", query);
+
+			if ((portal = SPI_cursor_open(NULL, plan, NULL, &nulls, true)) == NULL)
+				elog(ERROR, "SPI_cursor_open(\"%s\") failed", query);
+
+			/*
+			* According to specifictation, sp_volatility returns a result-set.
+			* If there is no destination, it will send the result-set to client, which is not allowed behavior of PG procedures.
+			* To implement this behavior, we added a code to push the result.
+			*/
+			receiver = CreateDestReceiver(DestRemote);
+			SetRemoteDestReceiverParams(receiver, portal);
+
+			/* fetch the result and return the result-set */
+			PortalRun(portal, FETCH_ALL, true, true, receiver, receiver, NULL);
+
+			receiver->rDestroy(receiver);
+			SPI_cursor_close(portal);
+
+			if ((rc = SPI_finish()) != SPI_OK_FINISH)
+				elog(ERROR, "SPI_finish failed: %s", SPI_result_code_string(rc));
+		}
+		PG_CATCH();
+		{
+			SPI_finish();
+			PG_RE_THROW();
+		}
+		PG_END_TRY();	
+	}
+	else
+	{	
+		volatility = lowerstr(volatility);
+		
+		if(strcmp(volatility,"volatile") && strcmp(volatility,"stable") && strcmp(volatility,"immutable"))
+			ereport(ERROR,
+					(errcode(ERRCODE_UNDEFINED_OBJECT),
+					errmsg("\"%s\" is not a valid volatility", volatility)));
+
+		query = psprintf("ALTER FUNCTION %s %s;", full_function_name, volatility);
```

Comment:
full_function_name cannot be greater than 128.  full_function_name = physical_schema_name.function_name and both physical_schema_name and function_name are 63 max after passing through `truncate_tsql_identifier`. Have added `truncate_tsql_identifier` in next commit

for volatility, it can be either 'stable', 'volatile' or 'immutable'. so maximum length that should be allowed is 9. Have added it in next commit.